doc/chapters/api/index.rst

```diff
@@ -3,150 +3,247 @@
+
+Error sources
+#############
+* Invalid ID: No matching container exists.
+* Bad state: The container is not ready or is suspended
+* Capability error: couldn't set default capabilities (if ``SetCapabilities`` has not already been
+  called)
+* LXC error: The underlying LXC method call failed.
+
+Note that even if the ``Execute`` call works fine, that doesn't mean the command that is being run
+inside the container runs fine. For example, it is possible to pass command-lines that point to
+non-executables, or non-existing files. One would notice this however, by getting a
+``ProcessStateChanged`` signal sent when the call exits.
 
 List
-----
+~~~~
 Returns a list of the current containers
 
-:Return Value:
-        :containers: ``array<int32>`` IDs for all containers
+Return value
+############
+* containers: ``array<int32>`` IDs for all containers
+
+Error sources
+#############
+None, this method only inspects the current state
 
 ListCapabilities
-----------------
-Lists all capabilities that the user can apply. Note that this does not include
-the default capabilities, which are not listable.
+~~~~~~~~~~~~~~~~
+Lists all capabilities that the user can apply. Note that this does not include the default
+capabilities, which are not listable.
+
+Return value
+############
+* capabilities: ``array<string>`` all available capability names
 
-:Return Value:
-        :capabilities: ``array<string>`` all available capability names
+Error sources
+#############
+None, this method only inspects the current state
 
 Resume
-------
+~~~~~~
 Resumes a suspended container
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
 
-|
+Error sources
+#############
+* Invalid ID: No matching container exists.
+* Bad state: The container is not suspended
+* LXC error: Couldn't resume the container
 
-SetCapabilities
----------------
-Applies the given list of capability names to the container. Capabilities are
-mapped to gateway configurations and applied to each gateway for which they
-map a configuration.
+Failure to resume a container leads to it being put in an invalid state.
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
-        :capabilities: ``array<string>`` of capability names
-
-|
+SetCapabilities
+~~~~~~~~~~~~~~~
+Applies the given list of capability names to the container. Capabilities are mapped to gateway
+configurations and applied to each gateway for which they map a configuration.
+
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+* capabilities: ``array<string>`` of capability names
+
+Error sources
+#############
+* Invalid ID: No matching container exists.
+* Bad state: The container is not ready
+* The given array of capabilities is empty (semantically not an error though)
```

Comment:
Yes, but then one doesn't call SetCapabilities at all, and the Execute call will check if any caps has been set. If not, it will set the default ones.